Composer/packages/ui-plugins/select-skill-dialog/src/SkillEndpointField.tsx

```diff
@@ -3,50 +3,48 @@
 
 /** @jsx jsx */
 import { jsx } from '@emotion/core';
-import React from 'react';
-import { FieldProps, useFormConfig } from '@bfc/extension-client';
-import {
-  getUiLabel,
-  getUIOptions,
-  getUiPlaceholder,
-  getUiDescription,
-  schemaField,
-  SelectField,
-} from '@bfc/adaptive-form';
+import React, { useMemo } from 'react';
+import { Dropdown, IDropdownOption } from 'office-ui-fabric-react/lib/Dropdown';
+import { FieldProps, useShellApi } from '@bfc/extension-client';
+import { FieldLabel } from '@bfc/adaptive-form';
+import { getSkillNameFromSetting, Skill } from '@bfc/shared';
 
 export const SkillEndpointField: React.FC<FieldProps> = (props) => {
-  const { depth, schema, uiOptions: baseUIOptions, value, onChange } = props;
-  const formUIOptions = useFormConfig();
+  const { description, label, required, uiOptions, value } = props;
+  const { shellApi, skillsSettings, skills = [] } = useShellApi();
+  const { updateSkillSetting } = shellApi;
 
-  const uiOptions = {
-    ...getUIOptions(schema, formUIOptions),
-    ...baseUIOptions,
-  };
+  const id = getSkillNameFromSetting(value);
+  const skill = skills.find(({ id: skillId }) => skillId === id) || ({} as Skill);
+  const { endpointUrl, msAppId } = skillsSettings[id] || {};
+
+  const { endpoints = [] } = skill;
 
-  const deserializedValue = typeof uiOptions?.serializer?.get === 'function' ? uiOptions.serializer.get(value) : value;
+  const options = useMemo(
+    () =>
+      endpoints.map(({ name, endpointUrl, msAppId }, key) => ({
```

Comment:
Isn't endpointUrl is shadowed by L19 endpointUrl?

Reply:
The `endpointUrl` on L19 is the current setting. On L25 we are creating the array of options from all the available endpoints which each have their own `endpointUrl`. The map function uses the `endpointUrl` from the mapper function's scope, which is the correct behavior.

Reply:
I understand they are different variables, but the one inside the mapper is shadowing the outer scope one, we currently don't have this on: https://eslint.org/docs/rules/no-shadow so take this comment as nit.